src/components/Profile.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+<div>
+  <div class="profile-header">
+      <h2>{{user.isVolunteer ? 'Volunteer' : 'Student' }} Profile</h2>
+  </div>
+  <div class="name">
+    <div id="firstname">
+      <span>First name:</span>
+      <span v-show="'firstname' != activeEdit">{{ user.firstname }}</span>
+      <input type="text" v-model="user.firstname" v-show="'firstname' == activeEdit">
+      <button @click="editField('firstname')">{{ fields['firstname'] }}</button>
```

Comment:
Prefer using dot syntax when a key is a string with no special characters: `fields.firstname` and `fields['different-name']`